pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -2243,6 +2243,176 @@ def is_hermitian(self):
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.IntegerComparator.matrix`
+
+        Args:
+            value (int): The value :math:`L` that the state's decimal representation is compared against.
+            control_wires (Union[Wires, Sequence[int], or int]): wires to place controls on
+            geq (bool): If set to `True`, the comparison made will be :math:`\sigma \geq L`. If `False`, the comparison
+                made will be :math:`\sigma < L`.
+
+        Returns:
+           tensor_like: matrix representation
+
+        **Example**
+
+        >>> print(qml.IntegerComparator.compute_matrix(2, [0,1]))
+        [[1. 0. 0. 0. 0. 0. 0. 0.]
+         [0. 1. 0. 0. 0. 0. 0. 0.]
+         [0. 0. 1. 0. 0. 0. 0. 0.]
+         [0. 0. 0. 1. 0. 0. 0. 0.]
+         [0. 0. 0. 0. 0. 1. 0. 0.]
+         [0. 0. 0. 0. 1. 0. 0. 0.]
+         [0. 0. 0. 0. 0. 0. 0. 1.]
+         [0. 0. 0. 0. 0. 0. 1. 0.]]
+        >>> print(qml.IntegerComparator.compute_matrix(2, [0,1], geq=False))
+        [[0. 1. 0. 0. 0. 0. 0. 0.]
+         [1. 0. 0. 0. 0. 0. 0. 0.]
+         [0. 0. 0. 1. 0. 0. 0. 0.]
+         [0. 0. 1. 0. 0. 0. 0. 0.]
+         [0. 0. 0. 0. 1. 0. 0. 0.]
+         [0. 0. 0. 0. 0. 1. 0. 0.]
+         [0. 0. 0. 0. 0. 0. 1. 0.]
+         [0. 0. 0. 0. 0. 0. 0. 1.]]
+        """
+
+        if value is None:
+            raise ValueError("The value to compare to must be specified.")
+
+        if isinstance(value, int):
+            paulix_mat = qml.PauliX.compute_matrix()
+
+            if geq:  # comparison is i >= value
+                if value > 2 ** len(control_wires) - 1:  # if value exceeds Hilbert space size
+                    mat = np.eye(2 ** (len(control_wires) + 1))
+
+                else:
+                    padding_left = value * 2
+                    paulix_padding = (2 ** (len(control_wires) + 1) - padding_left) // 2
+                    paulix_blocks = [paulix_mat for _ in range(paulix_padding)]
+                    mat = block_diag(np.eye(padding_left), *paulix_blocks)
+
+            else:  # comparison is i < value
+                if value == 0:  # if value exceeds Hilbert space size
+                    mat = np.eye(2 ** (len(control_wires) + 1))
+
+                else:
+                    padding_right = (2 ** (len(control_wires)) - value) * 2
+                    paulix_padding = (2 ** (len(control_wires) + 1) - padding_right) // 2
+                    paulix_blocks = [paulix_mat for _ in range(paulix_padding)]
+                    mat = block_diag(*paulix_blocks, np.eye(padding_right))
+
+        else:
+            raise ValueError(f"The compared value must be an int. Got {type(value)}.")
+
+        return mat
+
```

Comment:
Is there a decomposition for this operation?

Reply:
I'd have to think on it. Essentially, it _is_ just a bunch of multicontrolled X gates based on the `geq` condition and the `value`. For instance, if `value=3`and `geq=False`, then this would be equivalent to

`MultiControlledX([0,1,2], "00") @ MultiControlledX([0,1,2], "01") @ MultiControlledX([0,1,2], "10")`

Does this count?

Also, what are the benefits of having a decomposition defined?

Reply:
Thinking about this now, it should be feasible with classical control flow that constructs either an identity or a `qml.MultiControlledX` operation (unless missing sg).